doc/docs/modules/ROOT/pages/reading.adoc

```diff
@@ -0,0 +1,475 @@
+|_(empty)_
+|No
+
+4+|*Relationship Specific Options*
+
+|`relationship.nodes.map`
+|If true return `source` and `target` nodes as Map<String, String>, otherwise we flatten the properties by returning
+every single node property as column prefixed by `source` or `target`
+|`true`
+|No
+
+|`relationship.source.labels`
+|List of source node Labels separated by `:`
+|_(empty)_
+|Yes
+
+|`relationship.target.labels`
+|List of target node Labels separated by `:`
+|_(empty)_
+|Yes
+
+|===
+
+^*^ Just one of the options can be specified at the time.
+
+== Read Data
+
+Reading data from a Neo4j Database can be done in 3 ways:
+
+ * <<bookmark-read-query,Custom Cypher Query>>
+ * <<bookmark-read-node,Node>>
+ * <<bookmark-read-rel,Relationship>>
+
+[[bookmark-read-query]]
+=== Custom Cypher Query
+
+You can specify a Cypher query in this way:
+
+[source,scala]
+----
+import org.apache.spark.sql.{SaveMode, SparkSession}
+
+val spark = SparkSession.builder().getOrCreate()
+
+spark.read.format("org.neo4j.spark.DataSource")
+  .option("url", "bolt://localhost:7687")
+  .option("query", "MATCH (n:Person) WITH n LIMIT 2 RETURN id(n) as id, n.name as name")
+  .load()
+  .show()
+----
+
+.Result of the above code
+|===
+|id |name
+
+|0|John Doe
+|1|Jane Doe
+|===
+
+[NOTE]
+We recommend that individual property fields be returned, rather than returning  graph entity (node, relationship, and path) types.
+This best maps to Spark's type system and yields best results.
+So instead writing this `MATCH (p:Person) RETURN p` please write this: `MATCH (p:Person) RETURN id(p) as id, p.name as name`.
+If your query returns a graph entity please use the `labels` or `relationship` modes instead.
+
+The struct of the Dataset returned by the query is influenced by the query itself,
+in this particular context it could happen that the connector won't be able to sample the Schema from the query,
+in these cases we suggest trying with the option `schema.strategy` set to `string` as described <<bookmark-string-strategy,here>>.
+
+==== Schema
+The first 10 (or any number specified by the `schema.flatten.limit` option) results will be flattened and the schema will be created from those properties.
+
+[[bookmark-read-node]]
+=== Node
+
+You can read nodes by specifiying a single label, or multiple labels. Like so:
+
+.Single label
+[source,scala]
+----
+import org.apache.spark.sql.{SaveMode, SparkSession}
+
+val spark = SparkSession.builder().getOrCreate()
+
+spark.read.format("org.neo4j.spark.DataSource")
+  .option("url", "bolt://localhost:7687")
+  .option("labels", "Person")
+  .load()
+----
+
+.Multiple label
+[source,scala]
+----
+import org.apache.spark.sql.{SaveMode, SparkSession}
+
+val spark = SparkSession.builder().getOrCreate()
+
+spark.read.format("org.neo4j.spark.DataSource")
+  .option("url", "bolt://localhost:7687")
+  .option("labels", "Person:Customer:Confirmed")
+  .load()
+----
+
+[NOTE]
+Label list can be speficied both with starting color or without it:
```

Comment:
```suggestion
Label list can be specified both with starting colon or without it:
```

(Sorry about the drive-by comment. I just looked at the main README and it looked outdated talking about Neo4j 3 and Spark 2 but then I spotted this PR and I'm very happy. Thanks for the work on the 4.0 Connector!)